src/java.naming/share/classes/com/sun/naming/internal/NamingManagerHelper.java

```diff
@@ -198,18 +198,11 @@ static ObjectFactory getObjectFactoryFromReference(
         }
         // All other exceptions are passed up.
 
-        // Not in class path; try to use codebase
-        String codebase;
+        // Not in class path; loading of a factory from remote
+        // codebase is not supported
         if (clas == null &&
-                (codebase = ref.getFactoryClassLocation()) != null) {
-            try {
-                clas = helper.loadClass(factoryName, codebase);
-                // Validate factory's class with the objects factory serial filter
-                if (clas == null || !filter.test(clas)) {
-                    return null;
-                }
-            } catch (ClassNotFoundException e) {
-            }
+            ref.getFactoryClassLocation() != null) {
+            return null;
```

Comment:
This is what I suggest: 

at line 197 above (CNFE) just do:

```
        } catch (ClassNotFoundException e) {
            return null;
        }
```

Then you can remove the rest of the code that follows and simply do

```
        assert clas != null
        @SuppressWarnings("deprecation") // Class.newInstance
        ObjectFactory result = clas.newInstance();
        return result;
    }
```